test/tools/libtesteth/TestHelper.cpp

```diff
@@ -404,7 +404,75 @@ void checkCallCreates(eth::Transactions const& _resultCallCreates, eth::Transact
 	}
 }
 
-void executeTests(const string& _name, fs::path const& _testPathAppendix, fs::path const& _fillerPathAppendix, std::function<json_spirit::mValue(json_spirit::mValue const&, bool)> doTests, bool _addFillerSuffix)
+void executeTests2(const string& _name, fs::path const& _testPathAppendix, fs::path const& _fillerPathAppendix, test::TestSute const& _suite);
+void runAllTestsInFolder(test::TestSute const& _suite, const std::string& _folder)
+{
+	//testTypeParams params = getTestTypeParams(_type);
+	fs::path const fillersPath = fs::path(test::getTestPath()) / "src" / fs::path(_suite.folder() + "Filler") / fs::path(_folder);
+	string const filter = test::Options::get().singleTestName.empty() ? string() : test::Options::get().singleTestName + "Filler";
+	std::vector<boost::filesystem::path> const files = test::getJsonFiles(fillersPath.string(), filter);
+	size_t fileCount = files.size();
+	if (test::Options::get().filltests)
+		fileCount *= 2; //tests are checked when filled and after they been filled
+
+	fs::path const destTestFolder = fs::path(_suite.folder()) / fs::path(_folder);
+	fs::path const srcTestFolder = fs::path(_suite.folder() + "Filler") / fs::path(_folder);
+
+	auto testOutput = dev::test::TestOutputHelper(fileCount);
+	for (auto const& file: files)
+	{
+		test::TestOutputHelper::setCurrentTestFileName(file.filename().string());
+		executeTests2(file.filename().string(), destTestFolder.string(), srcTestFolder.string(), _suite);
+	}
+}
+
+void executeTests2(const string& _name, fs::path const& _testPathAppendix, fs::path const& _fillerPathAppendix, test::TestSute const& _suite)
```

Comment:
Well, we definitely should find a way to avoid this copy-paste. 
How about `runAllTestsInFolder` calling `executeTests` with lambda for `std::function` parameter
```
executeTests(...,
  [&_suite](json_spirit::mValue const& _input, bool _fillin) { _suite.doTests(_input, _fillin); } )
```
(this should be on line 425)

Reply:
I suppose this should be a temporary measure and in the end all test suites should implemet `TestSuite` and `executeTest` will have `TestSuite` parameter instead of `std::function` and lambda will be not needed

Reply:
I tried that it did not compile. 
this lambda function would have also to return json_spirit::mValue

Reply:
```
json_spirit::mValue suiteTestDo = [&_suite](json_spirit::mValue const& _input, bool _fillin)
	{
		return _suite.doTests(_input, _fillin);
	};

/home/wins/Ethereum/cpp-ethereum/test/tools/libtesteth/TestHelper.cpp:424: error: conversion from ‘dev::test::runAllTestsInFolder(const dev::test::TestSute&, const string&)::__lambda20’ to non-scalar type ‘json_spirit::mValue {aka json_spirit::Value_impl<json_spirit::Config_map<std::basic_string<char> > >}’ requested
  };
  ^
```

Reply:
sorry just checked out.  auto  lambda works!